### PR TITLE
memcpy requires valid pointers

### PR DESCRIPTION
**Pull request by ashtum**

### CODE REVIEW

**Comment by vinniefalco** [comment]
Suggested commit message: "memcpy requires a valid pointer"

**Comment by ashtum** [comment]
> Suggested commit message: "memcpy requires a positive size"

Technically the issue is passing null pointers.

File: include/boost/buffers/buffer_copy.hpp [hunk 1]
```diff
@@ -86,10 +86,11 @@ struct buffer_copy_impl
                     n = b1.size();
                 if( n > at_most - total)
                     n = at_most - total;
-                std::memcpy(
-                    b1.data(),
-                    b0.data(),
-                    n);
+                if(n != 0)
```
Comment (cmazakas) [inline include/boost/buffers/buffer_copy.hpp:89]: I think our style guide would have this written as `if( n != 0 )`

Comment (ashtum) [reply include/boost/buffers/buffer_copy.hpp:89]: `if(n != 0)` conforms to the style of other if statements in this repository. should I change it to `if( n != 0 )`?

**Pull request merged**